lib/easypost/services/address.rb

```diff
@@ -46,13 +46,6 @@ def all(filters = {})
 
   # Get the next page of addresses.
   def get_next_page(collection, page_size = nil)
-    # get_next_page_exec(method(:all), collection, collection.addresses, page_size)
-    params = {}
-    params[:before_id] = collection.addresses.last.id
-    unless page_size.nil?
-      params[:page_size] = page_size
-    end
-
-    @client.make_request(:get, 'addresses', MODEL_CLASS, params)
+    get_next_page_exec(collection, collection.addresses, 'addresses', MODEL_CLASS, page_size)
```

Comment:
issue: we need to rethink how this function works...

We are passing the following:
1. a collection of addresses
2. the unwrapped addresses
3. the string `addresses`
4. the model class, which is I assume `Address`

These are all basically the same thing. We should be able to pull out most of the necessary information by passing down much fewer parameters. Can we take a look at how this was done in other languages and make it act similarly? For instance, in python you pass the collection, an optional page size, and then other optional params.

Reply:
The extra parameters are just `MODEL_CLASS` and `unwrapped of addresses`. In other languages like Python, we are getting the unwrapped address automagically [here](https://github.com/EasyPost/easypost-python/blob/v7/easypost/resource.py#L81) and convert the response to the class automagically. In Ruby rewrite, we always pass in the class you are deserializing to. You have to call the method to grab the unwrapped addresses, so that's why it is explicitly grabbing the unwrapped parameter in the upper level.

Reply:
What I'm suggesting (and ultimately asking) is if there is a way we can simplify the interface for `get_next_page_exec`. Also, what does `exec` mean on that? Can we rename it to something more understandable?

Reply:
`exec` was from the previous name, I hate it too and will remove it. As of right now, not much to simplify unless we do automatically class deserialization in the `make_request` function or have some type of helper function(s) to grab the endpoint or the unwrapped parameter.

Reply:
I could take a crack at it, since I came up with the get_next_page concept (and therefore its overcomplexity is largely my fault).

Reply:
In my opinion, this is trivial, explicit vs automagical in the parameters. We can leave it until we finish the thread-safe and error-handling, then come back to this if we really want to optimize the number of parameters.